h2integrate/converters/grid/grid.py

```diff
@@ -0,0 +1,274 @@
+
+    def setup(self):
+        self.config = GridPerformanceModelConfig.from_dict(
+            merge_shared_inputs(self.options["tech_config"]["model_inputs"], "performance")
+        )
+
+        n_timesteps = self.options["plant_config"]["plant"]["simulation"]["n_timesteps"]
+
+        # Interconnection size input
+        self.add_input(
+            "interconnection_size",
+            val=self.config.interconnection_size,
+            units="kW",
+            desc="Maximum power capacity for grid connection",
+        )
+
+        # Electricity flowing INTO the grid (selling to grid)
+        self.add_input(
+            "electricity_in",
+            val=0.0,
+            shape=n_timesteps,
+            units="kW",
+            desc="Electricity flowing into grid interconnection point (selling to grid)",
+        )
+
+        # Electricity demand from downstream (for buying from grid)
+        self.add_input(
+            "electricity_demand",
+            val=0.0,
+            shape=n_timesteps,
+            units="kW",
+            desc="Electricity demand from downstream technologies",
+        )
+
+        # Electricity flowing OUT OF the grid (buying from grid)
+        self.add_output(
+            "electricity_out",
+            val=0.0,
+            shape=n_timesteps,
+            units="kW",
+            desc="Electricity flowing out of grid interconnection point (buying from grid)",
+        )
+
+        self.add_output(
+            "electricity_sold",
+            val=0.0,
+            shape=n_timesteps,
+            units="kW",
+            desc="Electricity sold to the grid",
+        )
+
+        self.add_output(
+            "electricity_unmet_demand",
+            val=0.0,
+            shape=n_timesteps,
+            units="kW",
+            desc="Electricity demand that is not met",
+        )
+
+        self.add_output(
+            "electricity_not_sold",
+            val=0.0,
+            shape=n_timesteps,
+            units="kW",
+            desc="Electricity that was not sold due to interconnection limits",
+        )
+
+    def compute(self, inputs, outputs):
+        interconnection_size = inputs["interconnection_size"]
+
+        # Selling: electricity flows into grid, limited by interconnection size
+        electricity_sold = np.clip(inputs["electricity_in"], 0, interconnection_size)
+        outputs["electricity_sold"] = electricity_sold
+
+        # Buying: electricity flows out of grid to meet demand, limited by interconnection
+        electricity_bought = np.clip(inputs["electricity_demand"], 0, interconnection_size)
+        outputs["electricity_out"] = electricity_bought
+
+        # Unmet demand if demand exceeds interconnection size
+        outputs["electricity_unmet_demand"] = inputs["electricity_demand"] - electricity_bought
+
+        # Not sold electricity if demand exceeds interconnection size
+        outputs["electricity_not_sold"] = inputs["electricity_in"] - electricity_sold
```

Comment:
can we unify the naming convention with other parts of H2I? In the battery we used `excess_{commodity}_out` I think. I see how that is confusing here, but multiple names can be confusing as well. I guess let's keep it as is unless you have a brilliant idea when you read this comment. Also mentioning #223 to track since I think we will likely keep the non-unified naming for now.

Reply:
Would `curtailed_{commodity}_out` be appropriate here, since it's a grid component?

Reply:
Based on our discussion on naming, we're using `<commodity>_excess` elsewhere. Would you prefer that here? I'll change it now.